packages/core/src/modules/authentication/use-cases/authenticate-with-github.ts

```diff
@@ -0,0 +1,43 @@
+import { z } from 'zod';
+
+type AuthenticateWithGithubInput = {
+  clientId: string;
+  clientSecret: string;
+  code: string;
+};
+
+const GithubOauthResponse = z.object({
+  access_token: z.string().trim().min(1),
+});
+
+type GithubOauthResponse = z.infer<typeof GithubOauthResponse>;
+
+/**
+ * Authenticates with GitHub using the `code` received from the OAuth flow.
+ * Returns the access token to be used with the GitHub API.
+ *
+ * @see https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps#2-users-are-redirected-back-to-your-site-by-github
+ */
+export async function authenticateWithGithub(
+  input: AuthenticateWithGithubInput
+) {
+  const response = await fetch('https://github.com/login/oauth/access_token', {
+    body: JSON.stringify({
+      client_id: input.clientId,
+      client_secret: input.clientSecret,
+      code: input.code,
+    }),
```

Comment:
I find the `JSON.stringify({})` option in addition to the `'Content-Type': 'application/json'` header a bit easier than creating/populating a `FormData` object.